server/authMiddleware.go

```diff
@@ -126,19 +130,29 @@ func getTokenInfo(token string, authServerURL string) (*User, error) {
 	if nil != e {
 		return nil, e
 	}
-	defer rs.Body.Close()
+	defer func() {
+		if err := rs.Body.Close(); err != nil {
+			log.Print(err)
+		}
+	}()
 
 	if rs.StatusCode/100 > 2 {
 		uatErr := new(UserInfoErr)
-		decodeJSON(rs, uatErr)
+		err := decodeJSON(rs, uatErr)
+		if nil != err {
+			return nil, err
+		}
 		e = &authError{
 			errorDesc:  uatErr,
 			statusCode: rs.StatusCode,
 		}
 		return nil, e
 	}
 	user := new(User)
-	decodeJSON(rs, user)
+	err := decodeJSON(rs, user)
+	if nil != err {
+		return nil, err
```

Comment:
please, use Logrus and Error level

Reply:
@Nolfgk

Reply:
not sure if i used it correctly, please check (i added log.Error here).